src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -39,6 +46,10 @@ public class GoblintAnalysis implements ServerAnalysis {
     private final GoblintClient goblintClient;
     private final GobPieConfiguration gobpieConfiguration;
     private final FileAlterationObserver goblintConfObserver;
+    private final int SIGINT = 2;
+    private static Future<?> lastAnalysisTask;
+    private static final ExecutorService execService = Executors.newSingleThreadExecutor();
+    private final AtomicBoolean analysisRunning = new AtomicBoolean(false);
```

Comment:
I didn't remove the `analysisRunning` variable, because `lastAnalysisTask` includes everything that GobPie does, including converting the results etc. Although these tasks should not really take too much time, I thought that as long as Goblint gets both aborted and killed with `SIGINT`, it would be safer to track the analysis run separately. So the `analysisRunning` gets updated right before and after writing to and reading from the socket, but the `lastAnalysisTask` keeps running until GobPie itself has finished with everything. So that is to prevent sending the `SIGINT` to Goblint when Goblint has finished the analysis, but GobPie hasn't yet finished converting the results for example.

I think we discussed with @vesalvojdani at some point that we should change Goblint to match "the standard" so that `SIGINT` doesn't kill it, but just gives `interrupted` when sent. But until then I would just in case keep it that way.